src/orange/orangeinp/detail/CsgUnitBuilder.cc

```diff
@@ -193,6 +194,47 @@ void CsgUnitBuilder::fill_volume(LocalVolumeId v,
     CELER_ENSURE(is_filled(unit_->fills[v.unchecked_get()]));
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Simplify negated joins for Infix evaluation.
+ *
+ * Apply DeMorgan simplification to use the \c CsgUnit in infix evaluation.
+ * \c NodeId indexing in the \c CsgTree are invalidated after calling this,
+ * \c CsgUnit data is updated to point to the simplified tree \c NodeId but any
+ * previously cached \c NodeId is invalid.
+ */
+void CsgUnitBuilder::simplifiy_joins()
+{
+    auto& tree = unit_->tree;
+    auto simplification = transform_negated_joins(tree);
+    CELER_EXPECT(tree.size() == simplification.new_nodes.size());
```

Comment:
It's not a precondition so:
```suggestion
    CELER_ASSERT(tree.size() == simplification.new_nodes.size());
```